dictdiffer/unify.py

```diff
@@ -24,19 +24,20 @@ def unify(self, first_patches, second_patches, conflicts):
         :param conflicts: list of Conflict objects
         """
         self.unified_patches = []
-        sorted_patches = sorted(first_patches + second_patches, key=get_path)
-
         self._build_index(conflicts)
 
+        sorted_patches = sorted(first_patches + second_patches, key=get_path)
+
         for patch in sorted_patches:
             conflict = self._index.get(nested_hash(patch))
 
+            # Apply only the patches that were taken as part of conflict
+            # resolution.
             if conflict:
-                if not conflict.handled:
-                    conflict.handled = True
```

Comment:
I removed this check since the same conflict can be checked twice:

* on its `first_patch`
* on its `second_patch`

Since we check below that the patch we are trying to apply is __either__ `first_patch` or `second_patch` ,given the conflict resolution, the second time the conflict is brought up from `self._index` it would always be not handled.